src/openrct2-ui/windows/Changelog.cpp

```diff
@@ -41,273 +41,300 @@ static constexpr const rct_string_id WINDOW_TITLE = STR_CHANGELOG_TITLE;
 constexpr int32_t MIN_WW = 300;
 constexpr int32_t MIN_WH = 250;
 
-static rct_widget window_changelog_widgets[] = {
+static rct_widget s_wndChangelogWidgets[] = {
     WINDOW_SHIM(WINDOW_TITLE, WW, WH),
     MakeWidget({0,  14}, {500, 382}, WindowWidgetType::Resize,      WindowColour::Secondary                               ), // content panel
     MakeWidget({3,  16}, {495, 366}, WindowWidgetType::Scroll,      WindowColour::Secondary, SCROLL_BOTH                  ), // scroll area
     MakeWidget({3, 473}, {300,  14}, WindowWidgetType::Placeholder, WindowColour::Secondary, STR_NEW_RELEASE_DOWNLOAD_PAGE), // changelog button
     { WIDGETS_END },
 };
 
-static void window_changelog_close(rct_window *w);
-static void window_changelog_mouseup(rct_window *w, rct_widgetindex widgetIndex);
-static void window_changelog_resize(rct_window *w);
-static void window_changelog_scrollgetsize(rct_window *w, int32_t scrollIndex, int32_t *width, int32_t *height);
-static void window_changelog_invalidate(rct_window *w);
-static void window_changelog_paint(rct_window *w, rct_drawpixelinfo *dpi);
-static void window_changelog_scrollpaint(rct_window *w, rct_drawpixelinfo *dpi, int32_t scrollIndex);
-
-static rct_window_event_list window_changelog_events([](auto& events)
-{
-    events.close = &window_changelog_close;
-    events.mouse_up = &window_changelog_mouseup;
-    events.resize = &window_changelog_resize;
-    events.get_scroll_size = &window_changelog_scrollgetsize;
-    events.invalidate = &window_changelog_invalidate;
-    events.paint = &window_changelog_paint;
-    events.scroll_paint = &window_changelog_scrollpaint;
-});
 // clang-format on
 
-static void window_new_version_process_info();
-static void window_changelog_dispose_data();
-static bool window_changelog_read_file();
-
-static const NewVersionInfo* _newVersionInfo;
-static std::vector<std::string> _changelogLines;
-static int32_t _changelogLongestLineWidth = 0;
-static int _persnality = 0;
-
-rct_window* window_changelog_open(int personality)
+class ChangelogWindow final : public Window
 {
-    rct_window* window;
-
-    window = window_bring_to_front_by_class(WC_CHANGELOG);
-    if (window != nullptr)
+private:
+    // Data
+
```

Comment:
You can remove those lines, class is private by default and we usually always have data at the top.